test/e2e/Makefile

```diff
@@ -0,0 +1,138 @@
+.DEFAULT_GOAL:=help
+
+IMG ?= localhost:5000/cluster-api-provider-cloudstack:latest
+
+GOPROXY := $(shell go env GOPROXY)
+ifeq ($(GOPROXY),)
+GOPROXY := https://proxy.golang.org
+endif
+export GOPROXY
+
+REPO_ROOT := $(shell git rev-parse --show-toplevel)
+
+help:
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Binaries
+## --------------------------------------
+
+BIN_DIR := $(REPO_ROOT)/bin
+GINKGO := $(BIN_DIR)/ginkgo
+KUSTOMIZE := $(BIN_DIR)/kustomize
+CONTROLLER_GEN := $(BIN_DIR)/controller-gen
+
+$(GINKGO): # Build ginkgo from bin folder.
+	@mkdir -p $(BIN_DIR)
+	GOBIN=$(BIN_DIR) go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+
+$(KUSTOMIZE): # Build kustomize from bin folder.
+	@mkdir -p $(BIN_DIR)
+	cd $(BIN_DIR) && curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash
+
+$(CONTROLLER_GEN): ## Install controller-gen to bin.
+	@mkdir -p $(BIN_DIR)
+	GOBIN=$(BIN_DIR) go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.9
+
+## --------------------------------------
+## Kind Cluster
```

Comment:
Same here. The main makefile already has a target for kind cluster.

Reply:
The reason why I didn't use kind-cluster target in the main Makefile is that it clones the entire cluster-api repo. In e2e test, it only needs to download kind-install-for-capd.sh file from the repo.

Reply:
I don't think we care about that though do we? The time it takes to pull down the cluster-api repo is seconds at most, and not that much data.

To me fewer code paths is better than saving the few megabytes of data transfer here.